github/client_repository_tree.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"context"
+	"strings"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/google/go-github/v45/github"
+)
+
+// TreeClient implements the gitprovider.TreeClient interface.
+var _ gitprovider.TreeClient = &TreeClient{}
+
+// TreeClient operates on the trees in a specific repository.
+type TreeClient struct {
+	*clientContext
+	ref gitprovider.RepositoryRef
+}
+
+// Create creates,updates,deletes a tree
+func (c *TreeClient) Create(ctx context.Context, tree *gitprovider.TreeInfo) (*gitprovider.TreeInfo, error) {
```

Comment:
We don't really have any tests for this function. Would that be easy to write? The other option would be to remove the `Create` for now as we don't really use it and make this a more Read-only (get/list) PR. wdyt?

Reply:
I think making it Read-only get/list is a good option, because the functionality needed so far from the create is implemented using the commit client where the tree is updated with the new tree structure